platform/android/app/src/main/kotlin/coop/polypoly/polypod/FeatureContainer.kt

```diff
@@ -122,6 +126,34 @@ class FeatureContainer(context: Context, attrs: AttributeSet? = null) :
             .show()
     }
 
+    private suspend fun approveEndpointFetch(
+        endpointId: String?,
+        completion: suspend (Boolean) -> String?
+    ): String? {
+        var fetchApproval: CompletableDeferred<Boolean?>? =
+            CompletableDeferred()
+        val featureName = feature?.name ?: throw Exception("endpoint failed")
```

Comment:
Errors are supposed to be little informative for the feature, we will have more informative errors in the backend

Reply:
Right, but a bit of more information is always welcome. Specifically, there are two exceptions with the same message. At least, it could be tailored to the specific problem that it highlights.

Reply:
We are trying to keep knowledge of the user's behaviour away from the feature. By making it opaque to the feature whether the user rejected the request or if, for example, the endpoint is not reaching back, will avoid the case in which the feature can implement some logic that will pester the user to convince them to accept the fetch request.

Reply:
@alexandrosFarhat Now that you introduced `PodApiError`, maybe create a specific one here to eliminate the duplicated message? Could still be something generic like "unable to contact endpoint".